src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -146,69 +180,122 @@ public void close() {
           "samService.verifyAuthorization");
 
-      int depth = (expand ? -1 : 1);
-
-      FSItem fsObject;
-      try {
-        String lookupTimer = performanceLogger.timerStart();
-        fsObject = fileService.lookupSnapshotFSItem(snapshotProject, drsId.getFsObjectId(), depth);
-
-        performanceLogger.timerEndAndLog(
-            lookupTimer,
-            drsObjectId, // not a flight, so no job id
-            this.getClass().getName(),
-            "fileService.lookupSnapshotFSItem");
-      } catch (InterruptedException ex) {
-        throw new FileSystemExecutionException(
-            "Unexpected interruption during file system processing", ex);
-      }
+      return lookupDRSObjectAfterAuth(expand, snapshot, drsObjectId, authUser, false);
+    }
+  }
 
-      if (fsObject instanceof FSFile) {
-        return drsObjectFromFSFile((FSFile) fsObject, drsId.getSnapshotId(), authUser);
-      } else if (fsObject instanceof FSDir) {
-        return drsObjectFromFSDir((FSDir) fsObject, drsId.getSnapshotId());
-      }
+  @VisibleForTesting
+  Snapshot lookupSnapshotForDRSObject(String drsObjectId) {
+    try {
+      DrsId drsId = drsIdService.fromObjectId(drsObjectId);
+      UUID snapshotId = UUID.fromString(drsId.getSnapshotId());
+      // We only look up DRS ids for unlocked snapshots.
+      String retrieveTimer = performanceLogger.timerStart();
+
+      Snapshot snapshot = snapshotService.retrieve(snapshotId);
+
+      performanceLogger.timerEndAndLog(
+          retrieveTimer,
+          drsObjectId, // not a flight, so no job id
+          this.getClass().getName(),
+          "snapshotService.retrieveAvailable");
+      return snapshot;
+    } catch (IllegalArgumentException ex) {
+      throw new InvalidDrsIdException("Invalid object id format '" + drsObjectId + "'", ex);
+    } catch (SnapshotNotFoundException ex) {
+      throw new DrsObjectNotFoundException(
+          "No snapshot found for DRS object id '" + drsObjectId + "'", ex);
+    }
+  }
+
+  void verifyPassportAuth(UUID snapshotId, DRSPassportRequestModel drsPassportRequestModel) {
+    SnapshotSummaryModel snapshotSummary = snapshotService.retrieveSnapshotSummary(snapshotId);
+    String phsId = snapshotSummary.getPhsId();
+    String consentCode = snapshotSummary.getConsentCode();
+    if (phsId == null || consentCode == null) {
+      throw new InvalidAuthorizationMethod("Snapshot cannot use Ras Passport authorization");
+    }
+    // Pass the passport + phs id + consent code to ECM
+    var criteria = new RASv1Dot1VisaCriterion().consentCode(consentCode).phsId(phsId);
+    criteria.issuer("https://stsstg.nih.gov").type("RASv1Dot1VisaCriterion");
+    var request =
+        new ValidatePassportRequest()
+            .passports(drsPassportRequestModel.getPassports())
+            .criteria(List.of(criteria));
+    var result = ecmService.validatePassport(request);
+
+    // log audit info
+    logger.info(result.getAuditInfo().toString());
```

Comment:
@nmalfroy  - Is this sufficient for the audit logging?

Reply:
Thanks for discussing offline. 
Example log:
`[Validate Passport Audit]: Data Repository accessed: TDR, Study/Data set accessed: <redacted>, Date/Time of access: 04-11-2022 16:02:40 EST, ECM Audit Info: {passport_jti=<redacted>, external_user_id=<redacted>, internal_user_id=<redacted>}`